src/textual/renderables/gradient.py

```diff
@@ -35,3 +39,90 @@ def __rich_console__(
                 ),
             )
             yield Segment(f"{width * ' '}\n", line_color)
+
+
+class LinearGradient:
+    """Render a linear gradient with a rotation."""
+
+    def __init__(self, angle: float, stops: list[tuple[float, Color]]) -> None:
+        """
+
+        Args:
+            angle: Angle of rotation in degrees.
+            stops: List of stop consisting of pairs of offset (between 0 and 1) and colors.
+        """
+        self.angle = angle
+        self._stops = stops[:]
+
+    def __rich_console__(
+        self, console: Console, options: ConsoleOptions
+    ) -> RenderResult:
+        width = options.max_width
+        height = options.height or options.max_height
+
+        angle_radians = -self.angle * pi / 180.0
+        sin_angle = sin(angle_radians)
+        cos_angle = cos(angle_radians)
+
+        center_x = width / 2
+        center_y = height
+
+        new_line = Segment.line()
+
+        color_gradient = Gradient(*self._stops)
+
+        _Segment = Segment
```

Comment:
Hangover from an iteration of the code?

Reply:
That's turning a global in to a local, which is faster. Use in L97